flink-ml-core/src/main/java/org/apache/flink/ml/common/broadcast/operator/AbstractBroadcastWrapperOperator.java

```diff
@@ -468,7 +425,9 @@ public void initializeState(StreamTaskStateInitializer streamTaskStateManager)
 
         timeServiceManager = streamOperatorStateContext.internalTimerServiceManager();
 
-        broadcastVariablesReady = false;
+        if (richContext.hasRichFunction) {
+            richContext.broadcastVariablesReady = false;
```

Comment:
Would it be more readable to explicitly initialize `broadcastVariablesReady` when `hasRichFunction == false`?

Reply:
Logically if `hasRichFunction == false`, we will never access the value of `broadcastVariablesReady`. So it seems unnecessary to set the value here.

But thanks for the catch, I found it unnecessary to call this function here since it is always initialized in `RichContext`. 

I will remove these three lines later.